starknet_py/tests/unit/signer/test_eth_signer.py

```diff
@@ -0,0 +1,159 @@
+            (0xEE8CC16FF762F7AF5B221AAA75177CC6, 0x4F6CFAB88C2AB909E4B80EBCEE151671),
+            0x0,
+        ),
+    ],
+    indirect=["load_typed_data"],
+)
+def test_message_hash(load_typed_data, expected_r, expected_s, expected_v):
+    eth_signer = EthSigner(
+        private_key=0x525BC68475C0955FAE83869BEEC0996114D4BB27B28B781ED2A20EF23121B8DE,
+        chain_id=StarknetChainId.MAINNET,
+    )
+    signed_message = eth_signer.sign_message(
+        load_typed_data,
+        0x65A822FBEE1AE79E898688B5A4282DC79E0042CBED12F6169937FDDB4C26641,
+    )
+
+    assert len(signed_message) == 5
+    assert signed_message[0] == expected_r[0]
+    assert signed_message[1] == expected_r[1]
+    assert signed_message[2] == expected_s[0]
+    assert signed_message[3] == expected_s[1]
+    assert signed_message[4] == expected_v
+
+
+@pytest.mark.asyncio
+async def test_validate_signature(
+    client, eth_account_class_hash, eth_fee_contract, strk_fee_contract
+):
+    signer = EthSigner(
+        0x525BC68475C0955FAE83869BEEC0996114D4BB27B28B781ED2A20EF23121B8DE,
+        chain_id=StarknetChainId.SEPOLIA,
+    )
+
+    # Manually serialize u512 into felt array
+    serializer = Uint256Serializer()
+    public_key_bytes = signer.public_key.to_bytes(64, byteorder="big")
+    constructor_calldata = serializer.serialize(
+        int.from_bytes(public_key_bytes[:32], byteorder="big")
+    ) + serializer.serialize(int.from_bytes(public_key_bytes[32:], byteorder="big"))
+
+    address, salt = _new_address(eth_account_class_hash, constructor_calldata)
+
+    await prepay_account(
+        address=address,
+        eth_fee_contract=eth_fee_contract,
+        strk_fee_contract=strk_fee_contract,
+    )
+
+    account = Account(
+        address=address,
+        client=client,
+        signer=signer,
+        chain=StarknetChainId.SEPOLIA.value,
+    )
+
+    deploy_account_tx = await account.sign_deploy_account_v3(
+        class_hash=eth_account_class_hash,
+        contract_address_salt=salt,
+        constructor_calldata=constructor_calldata,
+        resource_bounds=MAX_RESOURCE_BOUNDS,
+    )
+
+    await client.deploy_account(deploy_account_tx)
+
+    # Signature calculated first using starknet.js
+    # This is a call to the Map contract with
+    # address: 0x77AF8BE78F9AB70BA5DB594A079D53FB692256ACCA4F65ED4681BE63F766F8
+    # selector: put
+    # calldata: [100, 101]
+    call = Call(
+        to_addr=address,
+        selector=get_selector_from_name("is_valid_signature"),
+        calldata=[
+            0x78EDDEAD1D83CF611C1C78B488BC3CBFEFF0696E9464EA967676100BBC1E674,  # tx hash
+            0x5,
+            0x89B5C415C2F37F5A6D1292EDC15958B2,
+            0xA2EC923770F021309C58B7FA68A6A4CE,
+            0xC4AE697F7D4999C9C6F6C6F13C97EF65,
+            0x1792B1FB77858EE84EF3244807A73A16,
+            0x0,
+        ],
+    )
+    (res,) = await client.call_contract(call=call)
+    assert res == encode_shortstring("VALID")
```

Comment:
Not sure about the point of this test. We are deploying an account and then checking if a precalculated signature passes validation. Why not calculate the signature ourselves?